lib/usart/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! USART interface.
+//!
+//! This was formerly a driver and the implementation still looks largely
+//! driver-like in that it interacts with other drivers and system interrupts,
+//! but USARTs are inherently single-owner. Structuring USARTs as a lib allows
+//! the calling task to use the USART more directly than if it were a
+//! full-fledged driver of its own.
+
+#![no_std]
+
+#[cfg(any(feature = "stm32h743", feature = "stm32h753"))]
+pub mod stm32h7;
+
+#[cfg(any(feature = "stm32h743", feature = "stm32h753"))]
+use self::stm32h7::Device;
+
+use tinyvec::ArrayVec;
+
+/// Supported baud rates
+#[derive(Debug, Clone, Copy)]
+#[repr(u32)]
+pub enum BaudRate {
```

Comment:
What value is this enum adding? Could the user just pass in an integer? Otherwise I foresee us adding a bunch of baud rates here as applications develop.

Reply:
I was on the fence about this. I thought it might be nice to give the user constants to avoid typos, but I'm happy to remove it and take an integer instead.